packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
+    shorthand.as = 'button';
```

Comment:
I don't think it does https://github.com/microsoft/fluentui/blob/1ea12c61c6d47095f48d686b6a2ca9ab3bb98fd4/packages/react-aria/src/hooks/useAriaButton.ts#L65 does the check there

Reply:
No. that is actually event tested in the test file. if `shorhant.as==='button'` this hook pretty much does nothing. I just haven't shor circuited out because of internal hooks such as `useEventCallback`

Reply:
I'll try reordering hooks so that short-circuiting might be possible!